packages/node_modules/pouchdb-abstract-mapreduce/src/index.js

```diff
@@ -445,57 +448,61 @@ function createAbstractMapReduce(localDocName, mapper, reducer, ddocValidator) {
       return kvDocs;
     }
 
-    return getMetaDoc().then(function (metaDoc) {
-      return getKeyValueDocs(metaDoc).then(function (kvDocsRes) {
-        return processKeyValueDocs(metaDoc, kvDocsRes);
-      });
-    });
+    const metaDoc = await getMetaDoc();
+    const keyValueDocs = await getKeyValueDocs(metaDoc);
+    return processKeyValueDocs(metaDoc, keyValueDocs);
   }
 
   // updates all emitted key/value docs and metaDocs in the mrview database
   // for the given batch of documents from the source database
-  function saveKeyValues(view, docIdsToChangesAndEmits, seq) {
-    var seqDocId = '_local/lastSeq';
-    return view.db.get(seqDocId)
-      .catch(defaultsTo({_id: seqDocId, seq: 0}))
-      .then(function (lastSeqDoc) {
-        var docIds = mapToKeysArray(docIdsToChangesAndEmits);
-        return Promise.all(docIds.map(function (docId) {
-          return getDocsToPersist(docId, view, docIdsToChangesAndEmits);
-        })).then(function (listOfDocsToPersist) {
-          var docsToPersist = flatten(listOfDocsToPersist);
-          lastSeqDoc.seq = seq;
-          docsToPersist.push(lastSeqDoc);
-          // write all docs in a single operation, update the seq once
-          return view.db.bulkDocs({docs : docsToPersist});
-        });
-      });
+  async function saveKeyValues(view, docIdsToChangesAndEmits, seq) {
+    const seqDocId = '_local/lastSeq';
+    let lastSeqDoc;
+    try {
+      lastSeqDoc = await view.db.get(seqDocId);
+    } catch (err) {
+      if (err.status === 404) {
+        lastSeqDoc = {_id: seqDocId, seq: 0};
+      } else {
+        throw err;
+      }
+    }
+
+    const docIds = mapToKeysArray(docIdsToChangesAndEmits);
+    const listOfDocsToPersist = await Promise.all(docIds.map(function (docId) {
+      return getDocsToPersist(docId, view, docIdsToChangesAndEmits);
+    }));
+
+    const docsToPersist = flatten(listOfDocsToPersist);
+    lastSeqDoc.seq = seq;
+    docsToPersist.push(lastSeqDoc);
+    // write all docs in a single operation, update the seq once
+    return view.db.bulkDocs({docs : docsToPersist});
   }
 
   function getQueue(view) {
-    var viewName = typeof view === 'string' ? view : view.name;
-    var queue = persistentQueues[viewName];
+    const viewName = typeof view === 'string' ? view : view.name;
+    let queue = persistentQueues[viewName];
     if (!queue) {
       queue = persistentQueues[viewName] = new TaskQueue();
     }
     return queue;
   }
 
-  function updateView(view, opts) {
+  async function updateView(view, opts) {
     return sequentialize(getQueue(view), function () {
       return updateViewInQueue(view, opts);
     })();
   }
 
-  function updateViewInQueue(view, opts) {
+  async function updateViewInQueue(view, opts) {
```

Comment:
This function also currently doesn't need to be marked `async`, but we could change the code at the end of this function from `return createTask().then(function () {` onwards to use `await`. A lot of the internal structure of this function is very much designed around callbacks and can probably be simplified further but I wouldn't want to expand the scope of this PR right now.

Reply:
This was not addressed as we didn't want to expand the scope of the PR